### PR TITLE
[SPARK-42585][CONNECT][FOLLOWUP] Store cached local relations as proto

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to store the cached local relations in the proto format, the same as `LocalRelation`. Also I reverted `transformLocalRelation()` to the state before the commit https://github.com/apache/spark/commit/0d7618a2ca847cf9577659e50409dd5a383d66d3.

### Why are the changes needed?
To explicitly specify the format of cached local relations in the proto API.

### Does this PR introduce _any_ user-facing change?
Yes but the feature of cached local relations haven't been released yet.

### How was this patch tested?
By running the existing tests:
```
$ build/sbt "test:testOnly *.ArtifactManagerSuite"
$ build/sbt "test:testOnly *.ClientE2ETestSuite"
$ build/sbt "test:testOnly *.ArtifactStatusesHandlerSuite"
```

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala [hunk 1]
```diff
@@ -235,14 +233,14 @@ private[sql] class SparkConnectClient(
...
-    locRel.put(schemaBytes)
-    artifactManager.cacheArtifact(locRel.array())
+  def cacheLocalRelation(data: ByteString, schema: String): String = {
+    val localRelation = proto.Relation
+      .newBuilder()
+      .getLocalRelationBuilder
+      .setSchema(schema)
+      .setData(data)
+      .build()
+    artifactManager.cacheArtifact(localRelation.toByteArray)
```
Comment (amaliujia) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala:243]: is the byte format of the proto LocalRelation compatible with existing code? FYI @venkata91

Comment (amaliujia) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala:243]: Oh I see so from the server side it also re-construct LocalRelation proto from the bytes. Probably @venkata91 can confirm if all the places are updated.

Comment (vicennial) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala:243]: Thanks for the heads up! It looks good to me

**Review by vicennial — APPROVED** [review]

**Comment by MaxGekk** [comment]
The GA [Run / Build modules: pyspark-sql, pyspark-mllib, pyspark-resource](https://github.com/MaxGekk/spark/actions/runs/4934033324/jobs/8818623610#logs) frozen in the run https://github.com/MaxGekk/spark/runs/13365446543 but it passed in https://github.com/MaxGekk/spark/runs/13370236086

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @HyukjinKwon @vicennial @amaliujia for review.

**Pull request closed without merging**